recommender/recommender.py

```diff
@@ -963,6 +968,26 @@ def _get_statici18n_js_url():  # pragma: no cover
                 return text_js.format(lang_code=code)
         return None
 
+    def _get_statici18n_js_url(self):
+        """Return the JavaScript translation file provided by the XBlockI18NService."""
+        if url_getter_func := getattr(self.i18n_service, 'get_javascript_i18n_catalog_url', None):
+            if javascript_url := url_getter_func(self):
+                return javascript_url
+
+        if deprecated_url := self._get_deprecated_i18n_js_url():
+            return self.resource_string(deprecated_url)
+
+        return None
+
+    @property
+    def i18n_service(self):
+        """ Obtains translation service """
+        i18n_service = self.runtime.service(self, "i18n")
+        if i18n_service:
+            return i18n_service
+        else:
+            return DummyTranslationService()
```

Comment:
I'd love to know the goal of adding this function? Do we really need a `DummyTranslationService`?

If it's a copy from the LTI Consumer XBlock, I'm afraid we should remove it.